### PR TITLE
[SPARK-16988] [Spark Shell] spark history server log needs to be fixed to show https url when ssl is enabled

### PR DESCRIPTION
**Pull request by chie8842**
spark history server log needs to be fixed to show https url when ssl is enabled

### CODE REVIEW

**Comment by hayashidac** [comment]
Thank you for your comments. I’ll fix my program

**Comment by sarutak** [comment]
Hey @hayashidac , you did't need to close this PR. You should have just pushed your change...

**Comment by hayashidac** [comment]
I applied comments. Please check again.

File: core/src/main/scala/org/apache/spark/ui/WebUI.scala [hunk 1]
```diff
@@ -147,7 +147,10 @@ private[spark] abstract class WebUI(
   }
 
   /** Return the url of web interface. Only valid after bind(). */
-  def webUrl: String = s"http://$publicHostName:$boundPort"
+  def webUrl: String = {
+    val protocol = if (sslOptions.enabled) "https" else "http"
+      s"$protocol://$publicHostName:$boundPort"
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/ui/WebUI.scala]: Nit: don't indent this line further

**Comment by hayashidac** [comment]
sorry, I fixed it.

**Comment by sarutak** [comment]
Jenkins, test this please.

**Comment by sarutak** [comment]
While I I noticed that all of `spark.ui.ssl` in `SSLOptionsSuite.scala` should be `spark.ssl.ui`. It's very minor so you can fix it too within this PR if you would like to.

**Comment by srowen** [comment]
Looking good, thank you. I also wouldn't mind fixing the "spark.ui.ssl" -> "spark.ssl.ui" issue here because it's a little bit related.

**Comment by hayashidac** [comment]
I fixed SSLOptionsSuite.scala too.

**Review by srowen — APPROVED** [review]

**Comment by sarutak** [comment]
ok to test.

**Comment by sarutak** [comment]
LGTM. Merging into `master` and `branch-2`. Thanks @hayashidac .

**Pull request closed without merging**